st2api/st2api/controllers/resource.py

```diff
@@ -204,6 +206,22 @@ def _get_one_by_name_or_id(self, name_or_id, exclude_fields=None, from_model_kwa
 
         return result
 
+    def _get_one_by_pack_ref(self, pack_ref, exclude_fields=None, from_model_kwargs=None):
+        LOG.info('GET %s with pack_ref=%s', pecan.request.path, pack_ref)
+
+        instance = self._get_by_pack_ref(pack_ref=pack_ref, exclude_fields=exclude_fields)
```

Comment:
This can throw StackStormDBObjectNotFoundError. I am confused by the `if not instance` and abort. Actually I am confused more by why there is both _get_one_by_pack_ref and _get_by_pack_ref. Seems like there is a good scope for abuse.